test/bootboot_test.rb

```diff
@@ -209,6 +209,68 @@ def test_bootboot_command_initialize_the_next_lock_and_update_the_gemfile
     end
   end
 
+  def test_bundle_install_with_different_ruby_updating_gemfile_next_lock_succeeds
+    write_gemfile do |file, _dir|
+      FileUtils.cp("#{file.path}.lock", gemfile_next(file))
+      File.write(file, <<-EOM, mode: 'a')
+        if ENV['DEPENDENCIES_NEXT']
+          ruby '9.9.9'
+        else
+          ruby '#{RUBY_VERSION}'
+        end
+      EOM
+
+      run_bundler_command('bundle install', file.path)
+
+      assert_equal(
+        RUBY_VERSION,
+        Bundler::Definition.build(
+          file.path, "#{file.path}.lock", false
+        ).ruby_version.gem_version.to_s
+      )
+
+      error_message = <<~EOE
+        Running `bundle install` with a version of Ruby other than the one
+        specified in Gemfile_next.lock should update Gemfile_next.lock so that
+        gems are properly resolved using the Ruby version specified in the
+        respective lockfile regardless of the running Ruby version.
+      EOE
+
+      with_env_next do
+        assert_equal(
+          "9.9.9",
+          Bundler::Definition.build(
+            file.path, gemfile_next(file), false
+          ).ruby_version.gem_version.to_s,
+          error_message
+        )
+      end
+    end
+  end
+
+  def test_bundle_install_with_different_ruby_for_installing_gemfile_next_lock_fails
+    write_gemfile do |file, _dir|
+      FileUtils.cp("#{file.path}.lock", gemfile_next(file))
+      File.write(file, <<-EOM, mode: 'a')
+        if ENV['DEPENDENCIES_NEXT']
+          ruby '9.9.9'
+        else
+          ruby '#{RUBY_VERSION}'
+        end
+      EOM
+
+      error_message = <<~EOE
+        Running `DEPENDENCIES_NEXT=1 bundle install` with a version of Ruby
+        other than the one specified in Gemfile_next.lock should fail so that
+        gems are not installed under the wrong Ruby.
+      EOE
+
+      assert_raises BundleInstallError, error_message do
```

Comment:
Sorry my comment was probably not clear. I meant to make an assertion on the error message raised


```ruby
error = assert_raises BundleInstallError do
  run_bundler_command('bundle install', file.path, env: { Bootboot.env_next => '1' }
end

assert_equal("Your Ruby version is x.x.x, but your Gemfile specified x.x.x", error.message)
```

Reply:
Oh! No problem, will do.